crates/ruff/src/rules/flake8_logging/rules/exc_info_false_in_exception.rs

```diff
@@ -0,0 +1,59 @@
+use ruff_python_ast::ExprCall;
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::helpers::Truthiness;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for uses of `logging.exception()` with `exc_info` set to `False`
+///
+/// ## Why is this bad?
+/// The `exception()` method captures the exception automatically. Disabling this by setting
+/// `exc_info=False` is the same as using `error()`, which is clearer and doesn’t need the
+/// `exc_info`argument. This rule detects `exception()` calls with an exc_info argument that is
+/// falsy.
+///
+/// ## Example
+/// ```python
+/// logging.exception("foo", exc_info=False)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// logging.error("foo")
+/// ```
+#[violation]
+pub struct ExcInfoFalseInException;
+
+impl Violation for ExcInfoFalseInException {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Use of `logging.exception` with falsy `exc_info`")
+    }
+}
+
+/// LOG007
+pub(crate) fn exc_info_false_in_exception(checker: &mut Checker, call: &ExprCall) {
+    if checker
```

Comment:
This needs to catch all log messages.

More common than
```python
logging.exception("...")
```
is doing
```python
logger = logging.getLogger(__name__)

logger.exception("...")
```
ruff has utils to detect logging calls, see `is_logger_candidate` and e.g. the G001 implementation.

Reply:
Rewrote to utilise `is_logger_candidate`. It flags `logger.exception` and `logging.exception` now but does not flag `from logging import exception; exception()` Is that a limitation of `is_logger_candidate`? And if so, do we want to create another branch of logic to catch those cases?

Reply:
Moved this to https://github.com/astral-sh/ruff/issues/7502